src/daos/docker/test.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+if [[ $RUN_DATE == "" ]]; then
+    export RUN_DATE="$(date +"%Y-%m-%d")"
+fi
+if [[ $CONTAINER_NAME == "" ]]; then
+    export CONTAINER_NAME="dgws3-$RUN_DATE"
+fi
+set -x
+
+function start_docker_container()
+{
+    if [ ! "$(docker ps -q -f name=${CONTAINER_NAME})" ]; then
+        if [ "$(docker ps -aq -f status=exited -f name=${CONTAINER_NAME})" ]; then
+            # cleanup
+            docker rm ${CONTAINER_NAME}
+            if [[ ! $? == 0 ]]; then echo "failed"; exit 1; fi
+        fi
+        # run your container
+        docker run -it -d --privileged --cap-add=ALL -h docker-$CONTAINER_NAME --name $CONTAINER_NAME -v /dev:/dev dgw-s3-tests
+        if [[ ! $? == 0 ]]; then echo "failed"; exit 1; fi
+    fi
+
+    docker start $CONTAINER_NAME
+    if [[ ! $? == 0 ]]; then echo "failed"; exit 1; fi
+}
+
+function start_daos_cortx_s3tests()
+{
+    # check for anything running as this may be a re-run
+    PROCESSES=$(docker exec $CONTAINER_NAME  ps -e)
+    if [[ ! $PROCESSES =~ daos_server ]]; then
+        docker exec -u 0 $CONTAINER_NAME /opt/daos/bin/daos_server start &
+        if [[ ! $? == 0 ]]; then echo "failed"; exit 1; fi
```

Comment:
There is a lot of repeated code here, could be abstracted

Reply:
just the error handling.  How would you suggest this be modified?